support-frontend/assets/pages/supporter-plus-landing/components/countdown.tsx

```diff
@@ -0,0 +1,163 @@
+import { css } from '@emotion/react';
+import { headlineBold28, palette } from '@guardian/source/foundations';
+import { useEffect, useState } from 'react';
+import type { CountdownSetting } from 'helpers/campaigns/campaigns';
+/**
+ * This is used during the annual US End of Year Campaign.
+ */
+
+// create the style - TODO: design not yet confirmed...
+const gridStyle = css`
+	display: flex;
+	padding-top: 33px;
+	padding-bottom: 15px;
+	justify-content: center;
+	align-items: center;
+`;
+const gridItemStyle = css`
+	font-variant-numeric: lining-nums proportional-nums;
+	color: ${palette.brand[400]};
+	background-color: ${palette.neutral[100]};
+	/* 🖥 Headline/headline.bld.28 */
+	${headlineBold28}
+	/* font-family: 'headlineBold28';
+    font-size: 28px;
+    font-style: normal;
+    font-weight: 700; 
+    line-height: 115%;*/ /* 32.2px */
+    width: 65px;
+	margin: 3px;
+	padding: 2px;
+	border: 1px solid ${palette.brand[400]};
+	border-radius: 4px;
+	text-align: center;
+`;
+const timePortionStyle = css`
+	color: ${palette.brand[400]};
+	/* 🖥 Text Sans/text.sans.12 */
+	font-family: 'GuardianTextSans';
+	font-size: 12px;
+	font-style: normal;
+	font-weight: 400;
+	line-height: 130%; /* 15.6px */
+`;
+
+// props
+export type CountdownProps = {
+	campaign: CountdownSetting;
+};
+
+// create countdown logic
+const initialTimePart = '00';
+const millisecondsInSecond = 1000;
+const millisecondsInMinute = 60 * millisecondsInSecond;
+const millisecondsInHour = 60 * millisecondsInMinute;
+const millisecondsInDay = 24 * millisecondsInHour;
+
+// Questions:
+// handle inactive tabs via the Page Visibility API and calculate on visibility change?
+
+const ensureDoubleDigits = (timeSection: number): string => {
```

Comment:
is this function equivalent to `String(n).padStart(2, '0')`?

Reply:
Will adjust - better, thanks!